compiler.go

```diff
@@ -576,56 +729,84 @@ func (s *scope) finaliseVarAlloc(stackOffset int) (stashSize, stackSize int) {
 				}
 				code := scope.prg.code
 				base := scope.base
-				if argsInStash {
+				if this {
+					if derivedCtor {
```

Comment:
[opt] can combined this with the above check 

```
if this && derivedCtor{
...
}
```

Reply:
Both very legitimate changes! I'm only NOT going to do it to minimize conflicts in the future. Although our diversions are not clear yet (working on it ...) those are the only places where we can 100% make these type of changes. Much appreciate either way!